app/controllers/application_controller.rb

```diff
@@ -10,6 +10,16 @@ class ApplicationController < ActionController::Base
   before_action :set_locale
 
   rescue_from ActiveRecord::RecordNotFound, with: :render_404
+  before_action :set_csp
+
+  def set_csp
```

Comment:
> This seems pretty hacky/fragile.

I agree :+1: There are way two many options. I would like to suggest doing it with VCL if we can bare with keeping this logic out of the app.